plugins/modules/openssh_keypair.py

```diff
@@ -207,7 +220,7 @@ def generate(self, module):
             args = [
                 module.get_bin_path('ssh-keygen', True),
                 '-q',
-                '-N', '',
+                '-N', self.passphrase,
```

Comment:
This won't be merged as-is. Passprases must **never** be passed as command line parameter, they must always be passed via stdin.

(Also, you might be interested in #107, in particular: https://github.com/ansible-collections/community.crypto/issues/107#issuecomment-761848183 Also there's https://github.com/ansible/ansible/pull/46436#discussion_r222757521 Just so you know what kind of can of worms you are about to open ;-) )

Reply:
Lol, whoops.
Definitely don't write your secrets to the proc table. (Should've caught that)
Yeah, this module should probably be left alone unless someone wants to take inspiration from [builtin.user](https://github.com/ansible/ansible/blob/9ec4e08534209a2076023489c4a3f50f464d3b7a/lib/ansible/modules/user.py#L1148)
An additional module with a limited feature set not dependent on ssh-keygen could also be done using something like [this](https://pycryptodome.readthedocs.io/en/latest/src/public_key/rsa.html#Crypto.PublicKey.RSA.RsaKey.exportKey)
Thanks for the context and hopefully this will clarify the issue for anyone who is seeking this feature.

Reply:
Yes, avoiding ssh-keygen would be the best solution... (also for other reasons, like the issues in #107)

An alternative would be to use cryptography; it can load private and public keys:
* https://cryptography.io/en/latest/hazmat/primitives/asymmetric/serialization.html#openssh-public-key
* https://cryptography.io/en/latest/hazmat/primitives/asymmetric/serialization.html#openssh-private-key

It can also serialize some things so that OpenSSH should understand them:
* https://cryptography.io/en/latest/hazmat/primitives/asymmetric/serialization.html#cryptography.hazmat.primitives.serialization.PublicFormat.OpenSSH
* https://cryptography.io/en/latest/hazmat/primitives/asymmetric/serialization.html#cryptography.hazmat.primitives.serialization.PrivateFormat.OpenSSH
* https://cryptography.io/en/latest/hazmat/primitives/asymmetric/serialization.html#cryptography.hazmat.primitives.serialization.Encoding.OpenSSH

I've never used that parts of cryptography, so no idea how well it works, but in general cryptography is pretty good (also many modules in this collection use/support it).

Reply:
Yep, I was looking at cryptography as well.
Ultimately a partial or full implementation of the ssh-keygen feature set in python should probably only come from a security engineer to avoid the rabbit hole of security concerns. 
Also seems a little broader in scope than this project.